analytics/certificates-jan2025.md

```diff
@@ -0,0 +1,115 @@
+
+
+### Key registration
+
+The key registration records the public key and the proof of its possession.
+
+1. The Pool ID (or similar unique identifier) identifies the pool holding the key and comprises 32 bytes.
+2. The public key $\mathit{mvk}$ belongs to $\mathbb{G}_2$ , so it occupies 192 bytes if BLS12-381 is used.
+3. The proof of possession for the secret key is the pair $\left(H_{\mathbb{G}_1}(\text{``PoP''} \parallel \mathit{mvk})^\mathit{sk}, g_1^\mathit{sk}\right)$, where $\mathit{sk}$ is the secret key and $H$ hashes to points in $\mathbb{G}_1$. This pair will occupy 192 bytes.
+4. The KES signature for the above will add another 448 bytes.
+
+Altogether, a key registration occupies $32 + 192 + 192 + 448 = 864$ bytes.
+
+Sadly, this registration needs to be recorded on chain so that certificates can be later verified independently. However, cryptographic compression may reduce the overall size of the registration data.
+
+
+### Sortition
+
+Figure 7 of the [Fiat Accompli paper](https://iohk.io/en/research/library/papers/fait-accompli-committee-selection-improving-the-size-security-tradeoff-of-stake-based-committees/) provides the algorithm for determining which pools are persistent voters. The inequality for this determination can be computed exactly using rational arithmetic, so there is no danger of round-off errors. The input to the formula is the size of the committee and the distribution of stake among the pools.
+
+The non-persistent pools are subject to sortition for each vote, based on an updated stake distribution where the persistent voters have been removed and where the distribution is normalized to unit probability. The VRF value for that sortition is the bytes of the BLS signature on the election identifier $eid$. The probability that a pool with fraction $\sigma$ of the stake is awarded $k$ votes of the committee of $n$ votes is 
+
+$$
+\mathcal{P}(k) := \frac{(n \cdot \sigma)^k \cdot e^{- n \cdot \sigma}}{k!}
+$$
+
+This VRF value is used to look up the correct number of votes from the cumulative distribution for $\mathcal{P}(k)$. The same Taylor-series expansion technique used in Praos can handle the irrational arithmetic in a deterministic manner. In practice it is unlikely that the non-persistent pools would ever be awarded more than one vote, so it may be feasible to simply award one vote whenever $k \ge 1$: in such case the Peras voting sortition could use code very similar to the existing Praos block sortition.
+
+Each vote has a weight, with weights summing to unity. A quorum is achieved if the weights total to 0.75 or more.
+
+
+### Certification
+
+Consider the committee size $n$, which contains $m$ persistent voters.
+
+The certificate must contain the following information:
+
+- Common to all votes
+    - Election ID: Presumably a 32-byte identifier for the Peras election is included in the certificate, though perhaps this is not strictly necessary. This could just be the slot number.
```

Comment:
Any particular motivation for using 32 bytes? Slot numbers are currently always represented as 8 bytes.

Reply:
Fixed in 1bc140b560ca66522ccb6f4124f01df39dcb34ff.